### ISSUE TITLE
[Roadmap] Intel GPU (XPU) Backend Optimization (2025 H2)

### ISSUE THREAD

**Issue by mingfeima** [sgl-project/sglang#8309]
### Checklist

- [x] 1. If the issue you raised is not a feature but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [x] 2. Please use English, otherwise it will be closed.

### Motivation

Contribute to Intel GPU (XPU) backend to SGLang - Ref.No1 The Intel GPUs are abstracted as [`torch.xpu`](https://docs.pytorch.org/docs/stable/xpu.html) device in PyTorch. We target at the client GPU [Battlemage](https://newsroom.intel.com/client-computing/intel-launches-arc-b-series-graphics-cards) and workstation GPU [Battlematrix](https://www.intel.com/content/www/us/en/developer/articles/technical/introduction-project-battlematrix.html) at the current stage.

### Genrel Infrastructure
* **sgl-kernel-xpu**: general build and test infrastructure with sgl-kernel-xpu, https://github.com/sgl-project/sgl-kernel-xpu/pull/2

### General Operator Enabling
_(SyCL native Kernels with high performnace)_
* **Activation**: GELU， SiLU etc.
* **Normalization**: rmsnorm, gemma_rmsnorm, etc.
* **Embedding**: rotary embedding
* **GEMM**: bf16/f16 gemm for linear projection.
* **FP8 GEMM**: weight only quantization for FP8 E4M3 and E5M2.
* **INT4 AWQ/GPTQ**: weight only quantization for INT4 AWQ/GPTQ.
* **MoE**: fused MoE algo mapped to existing triton implementations.
* **Attention**: decode and extend attention for RadixAttention.

@airMeng @sunjiweiswift

### Parallelization
_(enable TP, DP, EP, etc)_
* **Tensor Parallel**: enable tensor parallel for multi-ARCs.

### Model Converage
Model Serial | Model Name | Model Size | Model Type | Input Type | Tensor Type | Data Type
-- | -- | -- | -- | -- | -- | --
Llama | Llama3.1-8B | 8.03B | Dense | Text | BF16 | INT4
Llama | Llama3.2-11B | 10.6B | Dense | Multimodal | BF16 | INT4
Llama | Llama3.3-70B | 70.6B | Dense | Text | BF16 | INT4
Llama | Llama-4-Scout-17B-16E | 109B | MoE | Multimodal | BF16 | INT4
Llama | Llama-4-Maverick-17B-128E-Instruct |   | MoE | Multimodal | BF16 | INT4
Qwen | Qwen3-14B | 14.8B | Dense | Text | BF16 | INT4
Qwen | Qwen3-32B | 32.8B | Dense | Text | BF16 | INT4
Qwen | Qwen3-30B-A3B | 30.5B | MoE | Text | BF16 | INT4
DeepSeek | DeepSeek-R1 | 685B | MoE | Text | FP8 | INT4
Phi | Phi-4-reasoning | 14.7B | Dense | Text | BF16 | INT4
Phi | Phi-4-multimodal-instruct | 5.57B | Dense | Multimodal | BF16 | INT4

@zhuyuhua-v

### Graph Mode
* Enable `torch.compile` graph mode for XPU device.

### Training (Fune Tuning)
* **lora**: lora and multi-lora enabling

### User Experience and Testing Enhancement
* Intel GPU CI Enabling and Testing @DiweiSun 
* Intel GPU documentation @ZailiWang @airMeng 

### Related resources

Ref.No1 - https://docs.pytorch.org/docs/stable/notes/get_start_xpu.html
